rust_src/src/lists.rs

```diff
@@ -262,6 +258,28 @@ impl<S: Into<LispObject>, T: Into<LispObject>> From<(S, T)> for LispObject {
     }
 }
 
+impl From<LispCons> for (LispObject, LispObject) {
+    fn from(c: LispCons) -> Self {
+        (c.car(), c.cdr())
+    }
+}
+
+impl From<LispObject> for (LispObject, LispObject) {
+    fn from(o: LispObject) -> Self {
+        LispCons::from(o).into()
+    }
+}
+
+impl From<LispObject> for Option<(LispObject, LispObject)> {
+    fn from(o: LispObject) -> Self {
```

Comment:
This would avoid checking if `o` is a LispCons twice. Is it better?

```
match Option<LispCons>::from(o) {
     Some(cell) => Some(cell.into()),
     None => None,
}
```